### ISSUE TITLE
Having trouble getting start - Rank newbie help please

### ISSUE THREAD

**Issue by MrYossu** [portacle/portacle#145]
I just ran Portacle (Windows 10) for the first time, as I'm interested in learning Lisp. However, I'm struggling to see how to use it for anything other than one-line programs.

I can enter something in the Slime buffer and click Return, and it's executed, but I don't seem to be able to edit code there. I guess I need to work out where I can enter and edit code, and then have it run. I'm sure this is hugely obvious, but as a first-time user, I just can't see where to start.

Please could someone point me in the right direction? Also, is there any help for Portacle? I couldn't see anything other than the (extremely) brief getting-started help on the [main web site page](https://portacle.github.io/).

Thanks for any help, and apologies for the really dumb question!

**Comment by Shinmera** [comment]
The buffer at the top says this:
```
;; For information on Portacle and how to use it, please read
;;   {url "https://portacle.github.io"} or {buffer "*portacle-help*"}
```

You should be able to click on `*portacle-help*`, which should open a buffer with a lengthy document on how to get started with Emacs and Portacle.

**Comment by MrYossu** [comment]
@Shinmera Thanks for the reply. I did look at those, but they didn't really help me. Maybe I'm being thick (it **is** Sunday after all!), but I couldn't see anything specific to Portacle. 

I have two immediate problems...

1) I can create a file and enter code, but can't see how to run that code in Portacle.

2) I am having problems editing the code. For example, if I enter the simple Lisp expression...

    (write(+ (* (/ 9 5) 60) 32))

...and then decide I want to change (say) the `60` to an expression, I don't seem to be able to do this. If I try to add an opening bracket before the `60`, Emacs adds a closing bracket straight after. If I delete the closing bracket (so I can put it after the `60`), it deletes the opening one as well.

As I said, I'm sure these are basic, dumb questions, but I'm struggling to get going here.

Thanks again.

**Comment by Shinmera** [comment]
1. https://github.com/portacle/config/blob/5e6314283e45c1bd45238c3b1400547954a7e77f/help.txt#L195-L196
   The typical workflow is to compile functions in your source files, and then hop to the REPL to run something when you want to test.
2. https://github.com/portacle/config/blob/5e6314283e45c1bd45238c3b1400547954a7e77f/help.txt#L213-L223
   Paredit can be confusing at first, but it's worth the learning curve. There's also an illustrated guide: http://danmidwood.com/content/2014/11/21/animated-paredit.html

**Comment by Shinmera** [comment]
Sorry I don't have time to write more elaborate explanations right now. I hope that helps to get somewhere at least.

**Comment by MrYossu** [comment]
OK, thanks for the replies. I can see that there is going to be a steep learning curve here even before I get into Lisp!

**Comment by Shinmera** [comment]
If Paredit is messing with you too much for now, you can also toggle it in a buffer by pressing `M-x paredit-mode`.

**Comment by MrYossu** [comment]
I think I have worked out how to stop it doing my head in so much. I can highlight text and then press `(` and it surrounds the highlighted text with brackets. Don't know if there is a better way, but this will do.

My bigger problem is working out how to run programs. If I have a Lisp program in a file, open in the top part of the window, how do I run it? The buffer that comes up in the lower part seems like a REPL, which is fine for entering small snippets, but how do I run a whole program in a file?

Thanks again.

**Comment by Shinmera** [comment]
You can use the Lisp function `LOAD` in the repl to load and run any file, or use something like `C-c C-k` to compile and load it. Also see https://common-lisp.net/project/slime/doc/html/Compilation.html

**Comment by MrYossu** [comment]
Thanks, that's a big help.

I just came across a couple of YouTube videos where people give an intro to Portacle, so between them, I think I'm ready to start learning Lisp!

Thanks again.

**Comment by Shinmera** [comment]
Good luck!

**Comment by MrYossu** [comment]
Thanks, I think I might need it!

I just got myself in a mess, where I couldn't type anything in the REPL, so closed Portacle and restarted. Now it comes up with this...

![Portacle](https://user-images.githubusercontent.com/10185332/114322365-bea57680-9b17-11eb-8c62-980a3d3953eb.png)

Any idea what I do now? I've restarted it a few times this afternoon, and never had this problem.

Thanks again.

**Comment by Shinmera** [comment]
Seems like some critical files were deleted. Extract the portacle distribution over your existing one.

**Comment by MrYossu** [comment]
Tried that, but it claimed the font files were in use by System. Had to restart the PC, delete the installation and reinstall. Not sure what happened, but I'm up and running again.

Let's hope it doesn't do that again!

**Comment by Shinmera** [comment]
Ah, the font thing is... unfortunate about Windows. The only way for portacle to ensure it has the fonts it needs  is to add them to the system, which in turn makes the files locked by the system until you reboot or manually free them through API calls.

**Comment by MrYossu** [comment]
Hmm, happened again. I was running a Lisp script, and Portacle seemed to stop responding, in that I couldn't type anything in any buffer, couldn't get any response out of any menu, etc. However, it wasn't using any CPU, and the window itself could be moved around. Clicking the red X in the top right corner didn't do anything.

I had to kill the process, and now I get the same error when I try to restart it. Can't re-install, the the copy fails when it gets to the fonts.

Isn't there some way around this? It seems very fragile.

Thanks again.

**Comment by Shinmera** [comment]
I'm afraid I've never encountered the issues you're reporting, so I don't know.

**Comment by MrYossu** [comment]
Is there a way of extracting the files manually? That way I could recopy what I need, and ignore the ones that are locked. As they are already there, I don't see why I would need to copy them again.

**Comment by Shinmera** [comment]
You can try opening the file with 7z, but I don't know if that'll work.

**Comment by MrYossu** [comment]
Thankfully it did. I extracted the files over the existing folder and it enabled me to start up again. 

Thanks for the help.

**Comment by MrYossu** [comment]
Famous last words, only worked that once.

Pretty much every time I close Portacle, next time I try to run it, I get the "Basic Portacle scripts are not present" message you see in the screenshot earlier (although not the rest of the message, as I created the `emacsd` folder manually). Deleting the old files (except for the fonts folder which won't delete until I restart) and copying the extracted files over doesn't help.

Any ideas? It's pretty unusable as it is.

**Comment by MrYossu** [comment]
@Shinmera  In case it helps anyone else, I think I have worked out what the problem was.

When I had it running, I pinned the icon to the taskbar, not realising that this pinned the Emacs executable, not the Portacle one. For some reason, running the Emacs executable directly gave errors. I Pinned portacle.exe to the taskbar, and now it works fine.

The downside to this is that I click the Portacle icon and then get the Emacs icon pop up, but that's no big deal.

Hope that helps someone else.
